tests/testthat/test-updating-state.R

```diff
@@ -254,3 +254,32 @@ test_that("Updating pars set initial state by default", {
   expect_equal(mod$pars(), new_pars)
   expect_equal(mod$state(), matrix(1, ncol = n_particles, nrow = 2))
 })
+
+test_that("Can reorder particles", {
```

Comment:
Some error handling tests missing here (these don't turn up in the coverage reports because we don't compile an example model into the package)

* test that providing out of range index errors (and leaves state unchanged)
* test that providing incorrect length index errors (& unchanged)

Also, 

* test that we can sample without replacement and duplication by passing in a vector like `c(5, 5, 2, 2, 1)` which should expose any issues in the temporary data storage